rollbar.go

```diff
@@ -31,53 +31,101 @@ const (
 	// All errors and messages will be submitted under this environment.
-	Environment = "development"
-
+	Environment string
 	// API endpoint for Rollbar.
-	Endpoint = "https://api.rollbar.com/api/1/item/"
-
+	Endpoint string
 	// Maximum number of errors allowed in the sending queue before we start
 	// dropping new errors on the floor.
-	Buffer = 1000
-
+	Buffer int
 	// Filter HTTP Headers parameters from being sent to Rollbar.
-	FilterHeaders = regexp.MustCompile("Authorization")
-
+	FilterHeaders *regexp.Regexp
 	// Filter GET and POST parameters from being sent to Rollbar.
-	FilterFields = regexp.MustCompile("password|secret|token")
-
+	FilterFields *regexp.Regexp
 	// String describing the running code version on the server
-	CodeVersion = ""
-
+	CodeVersion string
 	// host: The server hostname. Will be indexed.
-	ServerHost, _ = os.Hostname()
-
+	ServerHost string
 	// root: Path to the application code root, not including the final slash.
 	// Used to collapse non-project code when displaying tracebacks.
-	ServerRoot = ""
-
+	ServerRoot string
 	// Queue of messages to be sent.
 	bodyChannel chan map[string]interface{}
 	waitGroup   sync.WaitGroup
-)
+}
+
+type Client interface {
+	Error(level string, err error)
+	ErrorWithExtras(level string, err error, extras map[string]interface{})
+	ErrorWithStackSkip(level string, err error, skip int)
+	ErrorWithStackSkipWithExtras(level string, err error, skip int, extras map[string]interface{})
 
-// -- Setup
+	RequestError(level string, r *http.Request, err error)
+	RequestErrorWithExtras(level string, r *http.Request, err error, extras map[string]interface{})
+	RequestErrorWithStackSkip(level string, r *http.Request, err error, skip int)
+	RequestErrorWithStackSkipWithExtras(level string, r *http.Request, err error, skip int, extras map[string]interface{})
 
-func init() {
-	bodyChannel = make(chan map[string]interface{}, Buffer)
+	Message(level string, msg string)
+	MessageWithExtras(level string, msg string, extras map[string]interface{})
+
+	Wait()
+}
+
+func New(token, environment, codeVersion, serverHost, serverRoot string) Client {
+	return newRollbar(token, environment, codeVersion, serverHost, serverRoot)
+}
+
+func newRollbar(token, environment, codeVersion, serverHost, serverRoot string) *Rollbar {
+	buffer := 1000
+	client := &Rollbar{
+		Token:         token,
+		Environment:   environment,
+		Endpoint:      "https://api.rollbar.com/api/1/item/",
+		Buffer:        1000,
+		FilterHeaders: regexp.MustCompile("Authorization"),
+		FilterFields:  regexp.MustCompile("password|secret|token"),
+		CodeVersion:   codeVersion,
+		ServerHost:    serverHost,
+		ServerRoot:    serverRoot,
+		bodyChannel:   make(chan map[string]interface{}, buffer),
+	}
 
 	go func() {
-		for body := range bodyChannel {
-			post(body)
-			waitGroup.Done()
+		for body := range client.bodyChannel {
+			client.post(body)
+			client.waitGroup.Done()
 		}
 	}()
+	return client
+}
+
+var (
+	hostname, _ = os.Hostname()
```

Comment:
I just felt that querying `os.Hostname()` every time a Client is built was necessary. Thoughts?